example/example.go

```diff
@@ -28,7 +28,7 @@ func main() {
 
 		p := apns.NewPayload()
 		p.APS.Alert.Body = body
-		p.APS.Badge = &badge
+		p.APS.Badge = apns.NewBadgeNumber(badge)
```

Comment:
I think that I prefer the explicitness in tests (doesn't change field settings with changed implementations), and we don't expect users to initialize payloads this way.